pkg/deploy/spindeploy/transformer/telemetry.go

```diff
@@ -0,0 +1,111 @@
+package transformer
+
+import (
+	"context"
+	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha2"
+	"github.com/armory/spinnaker-operator/pkg/generated"
+	"github.com/armory/spinnaker-operator/pkg/inspect"
+	"github.com/armory/spinnaker-operator/pkg/version"
+	"github.com/ghodss/yaml"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	echoConfigFile     = "echo.yml"
+	kubernetesOperator = "kubernetes_operator"
+	telemetryKey       = "telemetry"
+)
+
+type telemetryTransformer struct {
+	log logr.Logger
+}
+
+type telemetryTransformerGenerator struct{}
+
+type DeploymentMethod struct {
+	DeploymentType    string `json:"type"`
+	DeploymentVersion string `json:"version"`
+}
+
+type Telemetry struct {
+	DeploymentMethod DeploymentMethod `json:"deploymentMethod"`
+}
+
+func (t *telemetryTransformerGenerator) NewTransformer(svc v1alpha2.SpinnakerServiceInterface,
+	client client.Client, log logr.Logger) (Transformer, error) {
+	tr := telemetryTransformer{log: log}
+	return &tr, nil
+}
+
+func (t *telemetryTransformerGenerator) GetName() string {
+	return "Telemetry"
+}
+
+func (t *telemetryTransformer) TransformConfig(ctx context.Context) error {
+	return nil
+}
+
+func (t *telemetryTransformer) TransformManifests(ctx context.Context, scheme *runtime.Scheme, gen *generated.SpinnakerGeneratedConfig) error {
+	for svc, cfg := range gen.Config {
+		if svc != "echo" {
+			continue
+		}
+		for k := range cfg.Resources {
+			sec, ok := cfg.Resources[k].(*v1.Secret)
```

Comment:
We unfortunately cannot offer any guarantee that this is the secret you want and you may have more than one. You should use `secretName := util.GetMountedSecretNameInDeployment(config.Deployment, k, "/opt/spinnaker/config")` followed by `sec := getSecretFromConfig(cfg.Resources[k], secretName)`.

Check out `accounts.go` for an example.